### PR TITLE
Exercise 4

### PR DESCRIPTION
**Pull request by PerryJones901**
Added Vagrant compatibility, so the web app can be run via Gunicorn inside a VM.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -118,3 +118,12 @@ env/
 ### Gecko Driver files ###
 geckodriver.log
 geckodriver.exe
+
+## Vagrant files ##
+.vagrant/
+
+## Poetry files ##
+poetry.lock
```
Comment (hugh-emerson) [inline .gitignore:126]: You should not ignore poetry.lock, even though it's auto-generated. You want to enforce specific versions

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+packaging = "20.4"
+pluggy = "0.13.1"
+py = "1.9.0"
+pylint = "2.5.2"
+pyparsing = "2.4.7"
+pytest = "5.4.3"
+python-dotenv = "0.12.0"
+requests = "2.24.0"
+rope = "0.17.0"
+selenium = "3.141.0"
```
Comment (hugh-emerson) [inline pyproject.toml:38]: You should move pytest, selenium etc into dev-dependencies section.

Also, you do not need to include nested dependencies. I expect you to be able to trim this list down to around 5 dependencies, plus two or three test ones.

**Review by hugh-emerson — APPROVED** [review]
Looks good. I've left some suggestions but merge when you're ready.

But you should update the README to mention the option of running the app via vagrant!

File: .gitignore (hunk 1 above)
Comment (PerryJones901) [reply .gitignore:126]: I have made this change on exercise-5 branch (which I'll make a PR for shortly)

File: pyproject.toml (hunk 2 above)
Comment (PerryJones901) [reply pyproject.toml:38]: I've also optimised this on exercise-5 branch too

**Pull request merged**